src/main/scala/edu/berkeley/ce/rockslicing/BlockVTK.scala

```diff
@@ -0,0 +1,197 @@
+      DenseMatrix.eye[Double](3)
+    }
+  }
+
+    /**
+   * Finds the center of a list of vertices - defined as the average coordinate of all the vertices in the list
+   * @param vertices List of vertices
+   * @return Coordinate of center of vertices
+   */
+  private def findCenter(vertices: Seq[(Double, Double, Double)]): (Double, Double, Double) = {
+    (vertices.map(_._1).sum / vertices.length.toDouble,
+      vertices.map(_._2).sum / vertices.length.toDouble,
+      vertices.map(_._3).sum / vertices.length.toDouble)
+  }
+
+  /**
+   * Compares pointA to pointB. If pointA is first relative to pointB rotating counter-clockwise about center,
+   * the method returns true. Input vectors should have same z-coordinate - comparison is based in 2-D
+   * @param pointA Point of interest
+   * @param pointB Point of comparison
+   * @param center Reference point for comparison
+   * @return Returns TRUE if pointA is first relative to pointB, FALSE otherwise
+   */
+  private def ccwCompare(pointA: (Double, Double, Double), pointB: (Double, Double, Double),
+                           center: (Double, Double, Double)): Boolean = {
+    assert(math.abs(pointA._3 - pointB._3) < BlockVTK.EPSILON)
+    if ((pointA._1 - center._1 < -BlockVTK.EPSILON) && (pointB._1 - center._1 >= BlockVTK.EPSILON)) {
+      return true
+    }
+    if ((pointA._1 - center._1 >= BlockVTK.EPSILON) && (pointB._1 - center._1 < -BlockVTK.EPSILON)) {
+      return false
+    }
+    if ((math.abs(pointA._1 - center._1) < BlockVTK.EPSILON) &&
+        (math.abs(pointB._1 - center._1) < BlockVTK.EPSILON)) {
+      if ((pointA._2 - center._2 >= BlockVTK.EPSILON) || (pointB._2 - center._2 >= BlockVTK.EPSILON)) {
+        return pointA._2 > pointB._2
+      }
+      return pointB._2 > pointA._2
+    }
+    // the cross product of vectors (pointA - center) and (pointB - center)
+    val det = (pointA._1 - center._1) * (pointB._2 - center._2) -
+      (pointB._1 - center._1) * (pointA._2 - center._2)
+    if (det > BlockVTK.EPSILON) {
+      return true
+    }
+    if (det < -BlockVTK.EPSILON) {
+      return false
+    }
+    // pointA and pointB are on the same line from the center, so check which one is closer to the center
+    val d1 = (pointA._1 - center._1) * (pointA._1 - center._1) + (pointA._2 - center._2) * (pointA._2 - center._2)
+    val d2 = (pointB._1 - center._1) * (pointB._1 - center._1) + (pointB._2 - center._2) * (pointB._2 - center._2)
+    return d1 > d2
+  }
+
+  /**
+   * Arranges the vertices of each of the faces of the input block such that they are in a counter-
+   * clockwise orientation relative their face's unit normal
+   * @param block A rock block
+   * @return A mapping from each face of the block to a Seq of vertices for that face arranged in counter-
+   *         clockwise order relative to its unit normal
+   */
+  private def orientVertices(block: Block): Map[Face, Seq[(Double, Double, Double)]] = {
+    val faces = block.findVertices
```

Comment:
It seems strange to name the result of `findVertices` something like `faces`. Either the variable or the method should be renamed so this makes more sense.